### PR TITLE
Feature/regions

### PR DESCRIPTION
**Pull request by Bodacious**
Changes proposed in this PR:
- Adds support for regions
- Provides database associations for regions and other content
- Upgrades Rails version
- Fixes minor typo bugs
- Refactors guidance group loading, from plans controller to view helpers
- Refactors relationship between Org and Plan
- Improve factory definitions

### CODE REVIEW

File: app/controllers/super_admin/themes_controller.rb [hunk 1]
```diff
@@ -51,7 +52,7 @@ def destroy
         redirect_to super_admin_themes_path, notice: msg
       else
         flash.now[:alert] = failure_message(@theme, _("delete"))
-        redner :edit
+        render :edit
```
Comment (briri) [inline app/controllers/super_admin/themes_controller.rb]: hah. good catch

File: app/helpers/plans_helper.rb [hunk 2]
```diff
@@ -1,5 +1,41 @@
+# frozen_string_literal
+
 module PlansHelper
 
+  def selected_guidance_groups_for_plan(plan)
```
Comment (briri) [inline app/helpers/plans_helper.rb]: Nice refactor. All of this logic made the Plan controller unwieldy

File: app/models/template.rb [hunk 3]
```diff
@@ -169,11 +168,11 @@ class Template < ActiveRecord::Base
   # default template)
   scope :latest_customizable, lambda {
     funder_ids = Org.funder.pluck(:id)
-    family_ids = families(funder_ids).distinct
-                                     .pluck(:family_id) + [default.family_id]
-    published(family_ids.uniq)
-      .where("visibility = :visibility OR is_default = :is_default",
-             visibility: visibilities[:publicly_visible], is_default: true)
+    family_ids = self.families(funder_ids).distinct.pluck(:family_id)
+    family_ids << 6912
```
Comment (briri) [inline app/models/template.rb]: this hard-coded id looks like its for debugging purposes?

File: spec/factories/regions.rb [hunk 4]
```diff
@@ -2,17 +2,14 @@
...
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
 #
 
 FactoryBot.define do
   factory :region do
-    name { Faker::Address.country }
-    abbreviation { SecureRandom.hex(2)  }
-    description { Faker::Lorem.sentence }
+    name { "MyString" }
```
Comment (briri) [inline spec/factories/regions.rb]: Why not use Faker here?

File: spec/features/questions/radiobuttons_questions_spec.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 require "rails_helper"
 
-RSpec.describe "Questions::Tadio button questions" do
+RSpec.describe "Questions::Radio button questions" do
```
Comment (briri) [inline spec/features/questions/radiobuttons_questions_spec.rb]: good catch

**Comment by johnpinto1** [comment]
@martaribeiro & @gjacob24 I am closing this but keeping the branch as suggested by @briri.

**Pull request closed without merging**